src/Module/Directory.php

```diff
@@ -0,0 +1,188 @@
+<?php
+
+namespace Friendica\Module;
+
+use Friendica\BaseModule;
+use Friendica\Content\Nav;
+use Friendica\Content\Pager;
+use Friendica\Content\Widget;
+use Friendica\Core\Hook;
+use Friendica\Core\L10n;
+use Friendica\Core\Renderer;
+use Friendica\Model\Contact;
+use Friendica\Model\Profile;
+use Friendica\Network\HTTPException;
+use Friendica\Util\Proxy as ProxyUtils;
+use Friendica\Util\Strings;
+
+/**
+ * Shows the local directory of this node
+ */
+class Directory extends BaseModule
+{
+	public static function init()
+	{
+		$app = self::getApp();
+
+		if (local_user()) {
+			$app->page['aside'] .= Widget::findPeople();
+			$app->page['aside'] .= Widget::follow();
+		} else {
+			unset($_SESSION['theme']);
+			unset($_SESSION['mobile-theme']);
+		}
+	}
+
+	public static function content()
+	{
+		$app = self::getApp();
+		$config = $app->getConfig();
+
+		if (($config->get('system', 'block_public') && !local_user() && !remote_user()) ||
+			($config->get('system', 'block_local_dir') && !local_user() && !remote_user())) {
+			throw new HTTPException\ForbiddenException();
```

Comment:
Please keep the former message like this:
```suggestion
			throw new HTTPException\ForbiddenException(L10n::t('Public access denied.'));
```